core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/OracleExportActionTest.java

```diff
@@ -0,0 +1,170 @@
+        return username.equals(USER) && password.equals(PASSPHRASE);
+      }
+    });
+    sshServer.setShellFactory(new ProcessShellFactory(SHELL_ARGS));
+    sshServer.setCommandFactory(EchoCommandFactory.INSTANCE);
+    return sshServer;
+  }
+
+  @Test
+  public void testCheckWrongArgumentForServerPassword() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, "wrongPassword", "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IOException e) {
+      Assert.assertEquals("SSH authentication error when connecting to test@" + sshServer.getHost() + " on port " +
+                            sshServer.getPort(), e.getMessage());
+    }
+  }
+
+  @Test
+  public void testCheckWrongArgumentForQuery() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, PASSPHRASE, "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IllegalArgumentException e) {
+      Assert.assertEquals("Query should have ; at the end", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testCheckWrongArgumentForFormat() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, PASSPHRASE, "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "format");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IllegalArgumentException e) {
+      Assert.assertEquals("Invalid format 'format'. Must be one of [csv, tsv, psv]", e.getMessage());
```

Comment:
better to construct this message in the same way as it is being constructed where the exception is thrown.